ISISNeutronMuon/MDANSE
# Restructured code and other small changes

**Description of work**
Restructured the code so that it follows the src-layout as shown in https://setuptools.pypa.io/en/latest/userguide/package_discovery.html which is more standard than the previous layout. Previously I was having some issues getting the editable build to work, the restructuring fixes this.  Also included some changes which improve things a bit which includes:

- updated gitignore to ignore: distribution/packaging files and directories,  generated pyd extensions, pycharm related stuff
- moved package data to the manifest.in
- changed script to entry-points so that they can be run as if they were executables
- updated the github wheels.yml to use python -m build since python setup.py bdist_wheel is deprecated https://packaging.python.org/en/latest/discussions/setup-py-deprecated/

**Fixes**
Fixes MDANSE and MDANSE_GUI installation using editable mode

**To test**
Run pip install -e . in the MDANSE and MDANSE_GUI and checked mdanse.exe and mdanse_gui.exe run without crashing. Built wheels with python -m build and installed them in a different environment to check building is done correctly and checked  mdanse.exe and mdanse_gui.exe again.
